cirq-core/cirq/work/sampler.py

```diff
@@ -329,7 +329,7 @@ def sample_expectation_values(
 
         # Flatten Circuit Sweep into one big list of Params.
         # Keep track of their indices so we can map back.
-        flat_params: List[Dict[str, float]] = [pr.param_dict for pr in study.to_resolvers(params)]
+        flat_params = [cast(Dict[str, float], pr.param_dict) for pr in study.to_resolvers(params)]
```

Comment:
Same here. We can add a cast / type ignore for now but also open a new issue to track the change. Ideally, the objects should be casted into `ParamDictType` instead of `Dict[str, float]`.

Reply:
Added #5282 to track